src/osmet/mod.rs

```diff
@@ -188,10 +188,19 @@ pub fn find_matching_osmet_in_dir(
 
 fn scan_root_partition(
     root: &Mount,
+    mapping: Option<&OsmetRootBlkDevReal>,
     mut boot_files: HashMap<u64, PathBuf>,
 ) -> Result<(OsmetPartition, HashMap<PathBuf, Sha256Digest>)> {
     // query the trivial stuff first
-    let (start_offset, end_offset) = root.get_partition_offsets()?;
+    let ((start_offset, end_offset), offset) = if let Some(mapping) = mapping {
+        let bdev = BlkDev { path: mapping.underlying_device.clone(), ..Default::default() };
+        let (start_offset, end_offset) = bdev.get_partition_offsets()?;
+        let offset = mapping.offset_bytes.checked_mul(512)
```

Comment:
Hmm, does this need to actually be `* $sector_size`? Or (like partition offsets), is it always in 512 blocks? We can punt on investigating this until we actually plumb through osmet for metal4k images.

Reply:
Good question; yeah, I think we should defer this until we've gone through the metal4k exercise.
I wouldn't be surprised if for 4k we also had to change the LUKS bits in RHCOS too.